src/CMakeLists.txt

```diff
@@ -867,7 +878,7 @@ endif(WITH_DOKAN)
 add_subdirectory(journal)
 
 if(WITH_RBD)
-  if(WITH_KRBD)
+  if(WITH_KRBD AND HAVE_KEYUTILS)
```

Comment:
I agree that in the context of this PR this change can be avoided due to KRBD being disabled. However, there is an unconditional linking with keyutils few lines below, so this addition seemed reasonable - unless you think that the correct behavior would be to issue an error if WITH_KRBD is set but HAVE_KEYUILS is off.

```
  if(WITH_KRBD AND HAVE_KEYUTILS)
    add_library(krbd STATIC krbd.cc
      $<TARGET_OBJECTS:parse_secret_objs>)
    target_link_libraries(krbd keyutils::keyutils)
  endif()
```

Reply:
> However, there is an unconditional linking with keyutils few lines below, so this addition seemed reasonable

There seems to be another instance of unconditional linking with keyutils in `src/mount/CMakeLists.txt`.  Why is a similar change not made there?

> unless you think that the correct behavior would be to issue an error if WITH_KRBD is set but HAVE_KEYUILS is off.

I would support that.  AFAICT it's been de-facto required for `krbd` and `mount.ceph` for many years.

Reply:
> There seems to be another instance of unconditional linking with keyutils in src/mount/CMakeLists.txt. Why is a similar change not made there?

The reason is that I first tried to fix the symptom by adding the keyutils condition, and then realized that KRBD should be disabled for Darwin, but didn't revert the keyutils change.

I will go ahead and add an error to the CMAKE requiring that KEYUTILS is present for when KRBD is enabled.

Reply:
> I will go ahead and add an error to the CMAKE requiring that KEYUTILS is present for when KRBD is enabled.

It would make sense to revert the keyutils change here and do the above in a separate PR.

Reply:
OK. Not ready for another PR at this time, but will revert this change here